xwiki-authenticator-trusted-api/src/main/java/org/xwiki/contrib/authentication/internal/DefaultTrustedAuthenticationConfiguration.java

```diff
@@ -83,11 +85,16 @@ public class DefaultTrustedAuthenticationConfiguration extends AbstractConfig
     private static final String PROPERTY_MAPPING_PROPERTY = "propertiesMapping";
     private static final char PROPERTY_MAPPING_SEP = '|';
 
+    private static final String DYNAMIC_ROLE_PROPERTY = "dynamicRole";
+    private static final String DYNAMIC_ROLE_CONFIGURATIONS_PROPERTY = DYNAMIC_ROLE_PROPERTY + ".configurations";
+    private static final String DYNAMIC_ROLE_CONFIGURATION_PREFIX = DYNAMIC_ROLE_PROPERTY + ".configuration.";
     private static final String LOGOUTPAGE_CONFIG_KEY = "xwiki.authentication.logoutpage";
 
     @Inject
     private ComponentManager componentManager;
 
+    private ArrayList<DynamicRoleConfiguration> dynamicRoleConfigurations;
```

Comment:
We generally use the base interface for cases like these, so here that would be List<DynamicRoleConfiguration>.

Reply:
Even for private field?

Just pure curiosity, I meant to use `Collection<DynamicRoleConfiguration>` anyway.

Reply:
Yes, it's just codestyle so not much to do with the location. In short, the rule is simply the fact that if you don't need more than what's exposed by `List<String>` you declare your variables with it instead of its specific implementation.

I remember seeing complains about that from our checkstyle, but maybe it was sonarlint.